### PR TITLE
feat: emitter parameters using godot custom properties

### PR DESCRIPTION
**Pull request by piiertho**
This adds fmod event parameters as properties of fmod event emitters.  
You can now set their value using classic inspector.  
You can also set them in script using custom godot property access, ex:
```gdscript
self["event_parameter/RPM/value"] = self["event_parameter/RPM/value"] + 10
```

### CODE REVIEW

File: demo/addons/fmod/tool/property_editors/FmodEventEditorProperty.gd [hunk 1]
```diff
@@ -0,0 +1,84 @@
+@tool class_name FmodEventEditorProperty extends FmodPathEditorProperty
+
+
+static var EVENT_PARAMETER_PREFIX_FOR_PROPERTIES = "event_parameter"
```
Comment (bitbrain) [inline demo/addons/fmod/tool/property_editors/FmodEventEditorProperty.gd:4]: caveat - using static var makes this addon incompatible with Godot 4.0 but only supports Godot 4.1 (we may want to reflect this in our compatibility matrix)

File: demo/addons/fmod/tool/property_editors/FmodEventEditorProperty.gd [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+static var fmod_parameter_ui: PackedScene = load("res://addons/fmod/tool/property_editors/FmodParameterUi.tscn")
+
+var former_event_description: FmodEventDescription
+
+func initialize(open_project_explorer_callable: Callable, path_prop: String, guid_prop: String):
```
Comment (bitbrain) [inline demo/addons/fmod/tool/property_editors/FmodEventEditorProperty.gd:9]: Why do we need this here?

File: demo/addons/fmod/tool/property_editors/FmodEventEditorProperty.gd [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	var event_description: FmodEventDescription = FmodServer.get_event_from_guid(get_edited_object().event_guid)
+	former_event_description = event_description
+
+func _set_path_and_guid(path: String, guid: String):
+	super(path, guid)
+	_update_parameters()
+	former_event_description = FmodServer.get_event_from_guid(get_edited_object().event_guid)
+
+func _update_parameters():
+	print("update params")
```
Comment (bitbrain) [inline demo/addons/fmod/tool/property_editors/FmodEventEditorProperty.gd:24]: Is this for debugging? Using `print_debug` might be more useful, as it gets omitted in Production builds.

File: demo/addons/fmod/tool/property_editors/FmodParameterUi.gd [hunk 4]
```diff
@@ -0,0 +1,28 @@
+@tool class_name FmodParameterUi extends VBoxContainer
+
+static var _step := 0.1
+
+func _ready():
+	%Slider.value_changed.connect(_on_slider_value_change)
```
Comment (bitbrain) [inline demo/addons/fmod/tool/property_editors/FmodParameterUi.gd:6]: same here - we should always cache nodes via `@onready var` for better performance.

**Comment by bitbrain** [comment]
Generally LGTM - just some minor feedback to improve GDScript performance + compatibility question!

File: demo/addons/fmod/tool/property_editors/FmodGuidAndPathPropertyEditorUi.gd [hunk 5]
```diff
@@ -0,0 +1,10 @@
+@tool class_name FmodGuidAndPathPropertyEditorUi extends HBoxContainer
+
+
+func set_callables(window_callable: Callable, path_callable: Callable, guid_callable: Callable):
+	%EventExplorerButton.pressed.connect(window_callable)
```
[the remark this answers is not in the document]
Comment (piiertho) [reply demo/addons/fmod/tool/property_editors/FmodGuidAndPathPropertyEditorUi.gd:5]: Here we cannot, it seems this method is called before ready.  
This is triggered by another function, triggered by editor.

File: demo/addons/fmod/tool/property_editors/FmodParameterUi.gd (hunk 4 above)
Comment (piiertho) [reply demo/addons/fmod/tool/property_editors/FmodParameterUi.gd:6]: dead code, will remove

File: demo/addons/fmod/tool/property_editors/FmodGuidAndPathPropertyEditorUi.gd (hunk 5 above)
[the remark this answers is not in the document]
Comment (bitbrain) [reply demo/addons/fmod/tool/property_editors/FmodGuidAndPathPropertyEditorUi.gd:5]: The problem is that if it is called before ready, `%EventExplorerButton` may not have been initialized yet, so there is a chance this code breaks. As long we are conscious about this it is fine.

[the remark this answers is not in the document]
Comment (piiertho) [reply demo/addons/fmod/tool/property_editors/FmodGuidAndPathPropertyEditorUi.gd:5]: Well all works well.
Got troubles if I pass to onready var.

File: demo/addons/fmod/tool/property_editors/FmodEventEditorProperty.gd (hunk 2 above)
Comment (piiertho) [reply demo/addons/fmod/tool/property_editors/FmodEventEditorProperty.gd]: removed

File: demo/addons/fmod/tool/property_editors/FmodEventEditorProperty.gd (hunk 1 above)
Comment (piiertho) [reply demo/addons/fmod/tool/property_editors/FmodEventEditorProperty.gd:4]: We do not use compatibility matrix anymore.  
We now release using this kind of tags: `x.y.z-X.Y.Z`, where `X.Y.Z` is godot version.

Comment (CedNaru) [reply demo/addons/fmod/tool/property_editors/FmodEventEditorProperty.gd:4]: If I remember Godot 4.1 extensions are not compatible with Godot 4.0 anyway, As Godot refactor their API in 4.1 in favor of a new "forward compitable" model.

**Review by CedNaru — APPROVED** [review]

**Pull request merged**